misk/src/main/kotlin/misk/moshi/okio/ByteStringAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package misk.moshi.okio
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import okio.ByteString
+import java.lang.reflect.Type
+
+/** JSON adapter converting [ByteString]s as base64 encoded strings */
+internal class ByteStringAdapter : JsonAdapter<ByteString>() {
+    override fun toJson(writer: JsonWriter, value: ByteString?) {
+        if (value != null) {
+            writer.value(value.base64Url())
+        } else {
+            writer.nullValue()
+        }
+    }
+
+    override fun fromJson(reader: JsonReader): ByteString? {
+        val nextToken = reader.peek()
+        return when (nextToken) {
+            JsonReader.Token.NULL -> reader.nextNull()
+            JsonReader.Token.STRING -> ByteString.decodeBase64(reader.nextString())
+            else -> throw IllegalArgumentException("expected base64 string, found $nextToken")
+        }
+    }
+
+    class Factory : JsonAdapter.Factory {
```

Comment:
Nit: no factory is necessary if it's just a simple type like this....

Oh but you do need it for the multibindings. Nevermind!